### PR TITLE
Fix some PHP 7.3 errors

### PR DESCRIPTION
**Pull request by bancer**

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks like only 3 errors left to undo the allow failure for php7.3 in travis matrix.

File: lib/Cake/Test/Case/Console/Command/Task/ExtractTaskTest.php [hunk 1]
```diff
@@ -469,10 +469,10 @@ public function testExtractModelValidationInPlugin() {
 		$this->Task->execute();
 		$result = file_get_contents($this->path . DS . 'test_plugin.pot');
 
-		$pattern = preg_quote('#Model/TestPluginPost.php:validation for field title#', '\\');
+		$pattern = preg_quote('#Model/TestPluginPost.php:validation for field title#', '/');
 		$this->assertRegExp($pattern, $result);
 
-		$pattern = preg_quote('#Model/TestPluginPost.php:validation for field body#', '\\');
+		$pattern = preg_quote('#Model/TestPluginPost.php:validation for field body#', '/');
```
Comment (garas) [inline lib/Cake/Test/Case/Console/Command/Task/ExtractTaskTest.php]: Remove `preg_quote` at all, it left from old versions when generated `.po` contained `\` on Windows.

**Comment by garas** [comment]
> PHP Warning:  PHP Startup: Unable to load dynamic library 'memcached.so' 

Disable `memcached` like in `3.x`
https://github.com/cakephp/cakephp/blob/d080e1d9a06d0461e86b415163cf0153b3496648/.travis.yml#L57-L60

**Comment by dereuromark** [comment]
```
There were 2 failures:
1) MultibyteTest::testUsingMbStrtoupper
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'ḀḂḄḆḈḊḌḎḐḒḔḖḘḚḜḞḠḢḤḦḨḪḬḮḰḲḴḶḸḺḼḾṀṂṄṆṈṊṌṎṐṒṔṖṘṚṜṞṠṢṤṦṨṪṬṮṰṲṴṶṸṺṼṾẀẂẄẆẈẊẌẎẐẒẔẖẗẘẙẚẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼẾỀỂỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪỬỮỰỲỴỶỸ'
+'ḀḂḄḆḈḊḌḎḐḒḔḖḘḚḜḞḠḢḤḦḨḪḬḮḰḲḴḶḸḺḼḾṀṂṄṆṈṊṌṎṐṒṔṖṘṚṜṞṠṢṤṦṨṪṬṮṰṲṴṶṸṺṼṾẀẂẄẆẈẊẌẎẐẒẔH̱T̈W̊Y̊AʾẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼẾỀỂỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪỬỮỰỲỴỶỸ'
/home/travis/build/cakephp/cakephp/lib/Cake/Test/Case/I18n/MultibyteTest.php:7634
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestCase.php:84
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestRunner.php:75
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestSuiteCommand.php:98
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:283
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:268
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Shell.php:462
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:219
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:66
2) MultibyteTest::testUsingMbStrtoupperArmenian
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'ԱԲԳԴԵԶԷԸԹԺԻԼԽԾԿՀՁՂՃՄՅՆՇՈՉՊՋՌՍՎՏՐՑՒՓՔՕՖև'
+'ԱԲԳԴԵԶԷԸԹԺԻԼԽԾԿՀՁՂՃՄՅՆՇՈՉՊՋՌՍՎՏՐՑՒՓՔՕՖԵՒ'
/home/travis/build/cakephp/cakephp/lib/Cake/Test/Case/I18n/MultibyteTest.php:7656
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestCase.php:84
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestRunner.php:75
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestSuiteCommand.php:98
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:283
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:268
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Shell.php:462
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:219
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:66
```

**Comment by bancer** [comment]
It is not clear why there are tests that assert the results of PHP functions such as `mb_strtoupper`. I see no point in that. Or should multibyte extension be removed on travis to test the framework's implementation of `mb_strtoupper`?

**Comment by markstory** [comment]
Looks pretty good to me. The multi byte class tests are a relic of how old 2.x is. Back when 2.x was started the mbstring extension was not commonly available. From what I remember the tests should be at least partially covering our code and not the just the php extension.

**Comment by bancer** [comment]
It seems there are some [changes](https://github.com/php/php-src/blob/php-7.3.0RC2/UPGRADING#L186) in mbstring in PHP 7.3 that affect these tests.

**Pull request merged**